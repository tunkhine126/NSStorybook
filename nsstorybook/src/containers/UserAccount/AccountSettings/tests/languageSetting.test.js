/**
 *
 * Tests for AccountSettings
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';

import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';

import LangSettings from '../languageSettings';

describe('<LangSettings />', () => {
  const data = {
    viewer: {
      language: 'en',
      defaultContextPermalink: 'us_en',
    },
  };

  const state = {
    survlangState: {
      item: 'us_en',
    },
    handleSurvLangChange: () => {},
    langState: { item: 'en' },
    handleLangChange: () => {},
    setSurvLangState: () => {},
    setLangState: () => {},
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <LangSettings data={data} state={state} />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <LangSettings data={data} state={state} />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
