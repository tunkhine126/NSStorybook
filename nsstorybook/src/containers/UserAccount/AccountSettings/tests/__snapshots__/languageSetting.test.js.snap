// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LangSettings /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    class="MuiGrid-root makeStyles-selectSpacing-216 MuiGrid-item"
  >
    <div
      class="NSSelect__ComponentStyle-sc-1pe1cg7-0 ikAthF"
    >
      <div
        class="MuiFormControl-root makeStyles-size-325"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
          for="default-survey-language-selector"
        >
          Default Survey Language
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby=" mui-component-select-default-survey-language"
            class="MuiSelect-root makeStyles-root-327 MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            id="mui-component-select-default-survey-language"
            role="button"
            tabindex="0"
          >
            us_en
          </div>
          <input
            id="default-survey-language-selector"
            name="default-survey-language"
            type="hidden"
            value="us_en"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-408 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-409"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <div
      class="NSSelect__ComponentStyle-sc-1pe1cg7-0 ikAthF"
    >
      <div
        class="MuiFormControl-root makeStyles-size-412"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
          for="default-language-selector"
        >
          Default Language
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby=" mui-component-select-default-language"
            class="MuiSelect-root makeStyles-root-414 MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            id="mui-component-select-default-language"
            role="button"
            tabindex="0"
          >
            English
            <strong
              class="makeStyles-infoKey-413"
            >
              (
              English
              )
            </strong>
          </div>
          <input
            id="default-language-selector"
            name="default-language"
            type="hidden"
            value="en"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-408 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-409"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;
